resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -97,28 +101,30 @@ public int negativeTtl() {
 
     @Override
     public void clear() {
-        for (Iterator<Map.Entry<String, List<DefaultDnsCacheEntry>>> i = resolveCache.entrySet().iterator();
+        for (Iterator<Map.Entry<String, Entries>> i = resolveCache.entrySet().iterator();
              i.hasNext();) {
-            final Map.Entry<String, List<DefaultDnsCacheEntry>> e = i.next();
+            final Map.Entry<String, Entries> e = i.next();
             i.remove();
-            cancelExpiration(e.getValue());
+
+            List<DefaultDnsCacheEntry> entryList = e.getValue().get();
+            if (entryList != null) {
+                cancelExpiration(entryList);
+            }
         }
     }
 
     @Override
     public boolean clear(String hostname) {
         checkNotNull(hostname, "hostname");
```

Comment:
I think I would keep it for now to make it clear we not support null at all.

Reply:
not a blocker but the decision to "not support null" could be consolidated in a single place (where we add to `resolveCache`) instead of spread out throughout the code.